plugins/couchbase/couchbaseembed.cpp

```diff
@@ -733,12 +746,28 @@ namespace couchbaseembed
     {
         if (script && *script)
         {
-            m_pQcmd = new Couchbase::QueryCommand(script);
+            // Incoming script is not necessarily null terminated. Note that the chars refers to utf8 characters and not bytes.
 
-            if ((m_scriptFlags & EFnoparams) == 0)
-                m_numParams = countParameterPlaceholders(script);
+            size32_t len = rtlUtf8Size(chars, script);
+
+            if (len > 0)
```

Comment:
Ah, but in an alternate universe where len < 0 the subsequent code would then fail.  I'm guarding against time-traveling extraterrestrials that use their own operating system.